src/main/java/sign_up/drivers/SignUpDatabaseAccess.java

```diff
@@ -0,0 +1,190 @@
+                String.valueOf(toSave.getId()), toSave.getPassword(), toSave.getEmail()};
+
+        try {
+            FileReader fileReader = new FileReader(file);
+            // create FileWriter object with file as parameter
+            CSVReader csvReader = new CSVReader(fileReader);
+            // create CSVWriter object filewriter object as parameter
+            List<String[]> csvBody = csvReader.readAll();
+            csvBody.add(toAdd);
+            //this is to get rid of extra line that gets added by the csvReader!
+            for(int i = 0; i < csvBody.size(); i++){
+                if(csvBody.get(i).length == 0 && csvBody.get(i)[0].equals("")){
+                    csvBody.remove(i);
+                    break;
+                }
+            }
+            FileWriter outputFile = new FileWriter(file);
+            CSVWriter writer = new CSVWriter(outputFile);
+            writer.writeAll(csvBody);
+            writer.flush();
+            // closing writer connection
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("Cannot find file");;
+        } catch (CsvException e) {
+            System.out.println("Incorrect File Format");
+        }
+
+    }
+
+    @Override
+    public String getAdminPassword() {
+        File file = new File(adminFilePath);
+        try {
+            // Create an object of fileReader
+            // class with CSV file as a parameter.
+            FileReader fileReader = new FileReader(file);
+
+            // create csvReader object passing
+            // file reader as a parameter
+            CSVReader csvReader = new CSVReaderBuilder(fileReader).withSkipLines(1).build();
+            String toReturn = csvReader.peek()[0];
+            csvReader.close();
+            return toReturn;
+        } catch (IOException e) {
+            System.out.println("Either wrong path or file has not been formatted correctly. ");
+            return "";
+        }
+    }
+
+    private ArrayList<String[]> readLineByLine(String filePath) throws IOException, CsvValidationException {
+        ArrayList<String[]> list = new ArrayList<>();
+        Reader reader = Files.newBufferedReader(Path.of(filePath));
+        CSVReader csvReader = new CSVReader(reader);
+        String[] line;
+        while ((line = csvReader.readNext()) != null) {
+            list.add(line);
+        }
+        return list;
+    }
+
+    private String convertListToString(ArrayList<Integer> list) {
+        String toReturn = "";
+
+        for (Integer integer: list) {
+            if (list.indexOf(integer) == 0) {
+                toReturn = integer.toString().concat(toReturn);
+            } else {
+                toReturn = integer.toString().concat(" ").concat(toReturn);
+            }
+        }
+        return toReturn;
+    }
+
+    private String convertDateToString(LocalDate toConvert) {
```

Comment:
You can use LocalDate.parse(String date) instead! The date has to be in the format yyyy-MM-dd though. It will automatically throw an error if the date does not exist, e.g. February 31st